File: packages/python/src/alumnium/tools/hover_tool.py
from pydantic import Field

from alumnium.drivers.base_driver import BaseDriver

from .base_tool import BaseTool


class HoverTool(BaseTool):
    """Hover over an element."""

    id: int = Field(description="Element identifier (ID)")

    def invoke(self, driver: BaseDriver):
        driver.hover(self.id)  # type: ignore[reportAttributeAccessIssue]  # ty:ignore[unresolved-attribute]

File: packages/java/src/main/java/ai/alumnium/tool/HoverTool.java
package ai.alumnium.tool;

import ai.alumnium.driver.BaseDriver;
import ai.alumnium.tool.annotation.ToolDescription;
import ai.alumnium.tool.annotation.ToolField;

@ToolDescription("Hover over an element.")
public record HoverTool(@ToolField(description = "Element identifier (ID)") int id)
    implements BaseTool {
  @Override
  public void invoke(BaseDriver driver) {
    driver.hover(id);
  }
}
